### DISCUSSION TITLE
可否实现不同widgets固定状态不同

### DISCUSSION THREAD

**Discussion by realzza** [ppoffice/hexo-theme-icarus#1167] [Q&A]
如题，希望实现不同widgets可以设定不同固定状态。

例如两栏状况下，profile card位于toc上方，希望profile card不固定，但是toc以及其他toc以下的widgets固定。有时目录过长，profile card会非常影响视线。谢谢。

**Comment by am-abudu** [comment]
这是我对于这个问题的修改，希望能对你有所帮助

`themes/icarus/source/js/main.js` 中的 `if ($toc.length > 0)` 后添加

```js
$toc.addClass('column-right is-sticky');   // 如果目录在左侧则改为 column-left
```

为了防止目录固定之后无法点击到较下方的项目，需要允许目录滚动

在 `themes/icarus/include/style/widget.styl` 添加

```scss
#toc  //目录可滚动
    max-height: calc(100vh - 22px)
    overflow-y: scroll
```

**Reply by PaRr0tBoY** [reply]
如果换成其他卡片呢？可否实现profile不动，在profile卡片下方的分类、归档、友链之类的卡片粘性固定
